pkg/cmd/params.go

```diff
@@ -0,0 +1,104 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// ServiceParams for creating integration and binding
+type ServiceParams struct {
+	AdditionalProperties bool                              `json:"additionalProperties"`
+	Properties           map[string]map[string]interface{} `json:"properties"`
+	Required             []string                          `json:"required"`
+	Type                 string                            `json:"type"`
+}
+
+func parseParams(keyVals []string) (map[string]string, error) {
+	params := map[string]string{}
+	for _, p := range keyVals {
+		kv := strings.Split(p, "=")
+		if len(kv) != 2 {
+			return nil, NewIncorrectParameterFormat("key value pairs are needed failed to find one: " + p)
+		}
+		params[strings.TrimSpace(kv[0])] = kv[1]
+	}
+	return params, nil
+}
+
+func isRequired(params ServiceParams, key string) bool {
+	for _, r := range params.Required {
+		if r == key {
+			return true
+		}
+	}
+	return false
+}
+
+// GetParams - Gets the service parameters (i.e. for provision/bind service) from the params
+//             flag or as a user input
+func GetParams(cmd *cobra.Command, params *ServiceParams) (*ServiceParams, error) {
```

Comment:
Do we need to pass in the full cobra.Command object here? or could we just pass in the stringArray?

Reply:
No, we can pass in the string array here as well :+1: Will update PR with these changes